artemis-core-client/src/main/java/org/apache/activemq/artemis/api/config/ActiveMQDefaultConfiguration.java

```diff
@@ -622,8 +622,8 @@ public static String getDefaultClusterPassword() {
    /**
     * This option controls whether passwords in server configuration need be masked. If set to "true" the passwords are masked.
     */
-   public static boolean isDefaultMaskPassword() {
-      return DEFAULT_MASK_PASSWORD;
+   public static Boolean isDefaultMaskPassword() {
```

Comment:
Why this?

Why you need a third state? true / false / undefined?

Why not just keep it boolean.. either mask it or not...


Besides.. if you really need the third option.. I would make DEFAULT_MASK_PASSWORD = null; instead of returning null here.


I couldn't understand why you need it.

Reply:
@clebertsuconic, check out org.apache.activemq.artemis.utils.PasswordMaskingUtil.resolveMask.  I believe the way that method works is why he needs 3 values.

Reply:
@clebertsuconic yes that's the purpose as @jbertram pointed out. I need 'null' to represent the case where the mask-password is not specified at all.